geode-core/src/main/java/org/apache/geode/pdx/internal/PeerTypeRegistration.java

```diff
@@ -350,23 +350,25 @@ private void releaseUDPMessaging(boolean release) {
   public int defineType(PdxType newType) {
     verifyConfiguration();
     Integer existingId = typeToId.get(newType);
+
     if (existingId != null) {
       return existingId;
     }
     lock();
     try {
-      int id = getExistingIdForType(newType);
-      if (id != -1) {
-        return id;
+      if (typeToId.isEmpty()) {
+        buildTypeToIdFromRegion();
+      }
+      // double check if my type is in region in case the typeToId map has been updated while
+      // waiting to obtain a lock
+      existingId = typeToId.get(newType);
```

Comment:
I think you need to make the same changes to this class in how it handles "enums". See defineEnum and enumToId.

Reply:
I purposely leave the enums untouched. It will be handled in another ticket.

Reply:
What is the ticket for the enum work?

Reply:
GEODE-7035